### PR TITLE
Add task dependency for existing task

### PR DESCRIPTION
**Pull request by vinit717**
**CLOSES** #1207 

image

### CODE REVIEW

File: models/tasks.js [hunk 1]
```diff
@@ -17,16 +17,34 @@ const { OLD_ACTIVE, OLD_BLOCKED, OLD_PENDING, OLD_COMPLETED } = TASK_STATUS_OLD;
...
+            const taskDependOn = {
+              taskId: taskId,
+              dependsOn: dependency,
+            };
+            const docid = dependencyModel.doc();
+            batch.set(docid, taskDependOn);
+          }
+        }
+        await batch.commit();
+      }
```
Comment (RitikJaiswal75) [inline models/tasks.js:46]: there are no tests for this, the coverage is not showing this as this is in the very beginning and this gets truncated when uncovered lines are displayed.

**Review by prakashchoudhary07 — DISMISSED** [review]

File: test/integration/tasks.test.js [hunk 2]
```diff
@@ -299,12 +299,63 @@ describe("Tasks", function () {
...
+        startedOn: 4567,
+        status: "COMPLETED",
+        dependsOn: ["taskId2", "taskId3"],
+        percentCompleted: 100,
+        participants: [],
+        assignee: appOwner.username,
+        completionAward: { [DINERO]: 3, [NEELAM]: 300 },
+        lossRate: { [DINERO]: 1 },
+        isNoteworthy: true,
+      };
```
Comment (heyrandhir) [inline test/integration/tasks.test.js:326]: it would be better if this can come from a fixture

**Review by heyrandhir — CHANGES_REQUESTED** [review]

File: test/integration/tasks.test.js (hunk 2 above)
Comment (vinit717) [reply test/integration/tasks.test.js]: done

File: test/integration/tasks.test.js [hunk 3]
```diff
@@ -299,12 +299,63 @@ describe("Tasks", function () {
...
+      //   startedOn: 4567,
+      //   status: "COMPLETED",
+      //   dependsOn: ["taskId2", "taskId3"],
+      //   percentCompleted: 100,
+      //   participants: [],
+      //   assignee: appOwner.username,
+      //   completionAward: { [DINERO]: 3, [NEELAM]: 300 },
+      //   lossRate: { [DINERO]: 1 },
+      //   isNoteworthy: true,
+      // };
```
Comment (heyrandhir) [inline test/integration/tasks.test.js:326]: can you remove the commented code

**Review by heyrandhir — CHANGES_REQUESTED** [review]

File: test/integration/tasks.test.js (hunk 3 above)
Comment (vinit717) [reply test/integration/tasks.test.js]: done

**Review by heyrandhir — APPROVED** [review]

**Pull request merged**